core/src/main/kotlin/org/evomaster/core/search/gene/collection/ArrayGene.kt

```diff
@@ -257,6 +268,51 @@ class ArrayGene<T>(
                 closingTag
     }
 
+    override fun getDtoCall(dtoName: String, counter: Int): DtoCall {
+        val result = mutableListOf<String>()
+
+        val listType = getListType(dtoName,template)
+        val listVarName = "list_${listType}_${counter}"
+        result.add("List<$listType> $listVarName = new ArrayList<$listType>();")
+
+        if (template is ObjectGene) {
+            val childDtoName = StringUtils.capitalization(template.refType?: name)
+            var listCounter = 1
+            elements.forEach {
+                val childDtoCall = it.getDtoCall(childDtoName, listCounter++)
+                result.addAll(childDtoCall.objectCalls)
+                result.add("$listVarName.add(${childDtoCall.varName});")
+            }
+        } else {
+            elements.forEach {
+                result.add("$listVarName.add(${it.getValueAsPrintableString(targetFormat = null)});")
+            }
+        }
+
+        return DtoCall(listVarName, result)
+    }
+
+    // this should actually be something we can ask the gene, like gene.getDtoType()
```

Comment:
What are your thoughts regarding this comment?

Reply:
putting inside Gene might be tricky, as it only apply to some genes, not all. also this is very specific  for DTO-based problems. i think "for now" this function could go in a utility class for DTO (and not inside `ArrayGene`). future might want to refactor it (eg inside `Gene`), but unsure now